src/main/java/com/coveros/selenified/services/HTTP.java

```diff
@@ -255,47 +258,24 @@ public Response delete(String service, Request request) {
      * @return Response: the response provided from the http call
      */
     private Response call(String call, String service, Request request, File file) {
-        StringBuilder params = new StringBuilder();
-        if (request != null && request.getParams() != null) {
-            params.append("?");
-            for (String key : request.getParams().keySet()) {
-                params.append(key);
-                params.append("=");
-                params.append(request.getParams().get(key));
-                params.append("&");
-            }
-        }
         HttpURLConnection connection = null;
         try {
-            URL url = new URL(this.serviceBaseUrl + service + params.toString());
+            URL url = new URL(this.serviceBaseUrl + service + getRequestParams(request));
             connection = (HttpURLConnection) url.openConnection();
-            String method = call;
-            if (PATCH.equals(method)) {
-                connection.setRequestProperty("X-HTTP-Method-Override", PATCH);
-                method = "POST";
-            }
-            connection.setRequestMethod(method);
-            connection.setRequestProperty("Content-length", "0");
-            connection.setRequestProperty(CONTENT_TYPE, contentType);
-            connection.setRequestProperty("Accept", "application/json");
-            for (Map.Entry<String, String> entry : extraHeaders.entrySet()) {
-                connection.setRequestProperty(entry.getKey(), entry.getValue());
-            }
-            connection.setDoOutput(true);
-            connection.setDoInput(true);
-            connection.setUseCaches(false);
-            connection.setAllowUserInteraction(false);
+            connection.setRequestMethod(call);
+            connection = setupHeaders(connection);
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Remove this useless assignment; "connection" already holds the assigned value along all execution paths. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](http://localhost:9000/sonar/coding_rules#rule_key=squid%3AS4165)